modules/repository/repo.go

```diff
@@ -80,6 +80,12 @@ func MigrateRepositoryGitData(ctx context.Context, u *user_model.User,
 		return repo, fmt.Errorf("Clone: %v", err)
 	}
 
+	if git.CheckGitVersionAtLeast("2.18") == nil {
+		if _, err := git.NewCommand(ctx, "commit-graph", "write").RunInDir(repoPath); err != nil {
```

Comment:
This will likely represent a huge performance improvement across the board.

We need to force write the commit graph on run sync too